scipy/optimize/optimize.py

```diff
@@ -3217,6 +3217,24 @@ def show_options(solver=None, method=None, disp=True):
     - :ref:`simplex         <optimize.linprog-simplex>`
     - :ref:`interior-point  <optimize.linprog-interior-point>`
 
+    Examples
+    --------
+    We can print documentations of a solver in stdout:
+
+    >>> from scipy.optimize import show_options
+    >>> show_options(solver="minimize")
```

Comment:
This generates output, so the refguide checker will complain about it missing. I suggest to show the first couple of lines of output, followed by `...`. 

You'll also have to add it to `DOCTEST_SKIPLIST` in `tools_refguide_check.py` I think, for CI to not break.